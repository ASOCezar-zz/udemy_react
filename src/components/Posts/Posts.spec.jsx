import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'title 1',
      body: 'body 1',
      cover: 'img/img1',
    },
    {
      id: 2,
      title: 'title 2',
      body: 'body 2',
      cover: 'img/img2',
    },
    {
      id: 3,
      title: 'title 3',
      body: 'body 3',
      cover: 'img/img3',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts on screen', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { img: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  it('should not render posts', () => {
    const posts = [];
    render(<Posts posts={posts} />);

    expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    const { container } = render(<Posts {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
