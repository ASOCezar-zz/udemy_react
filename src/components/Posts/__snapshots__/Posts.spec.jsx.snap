// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match with snapshot 1`] = `
<div
  class="sc-gtsrHT Yrjky posts"
>
  <div
    class="sc-bdnxRM bkXcNm post"
  >
    <img
      alt="title 1"
      src="img/img1"
    />
    <div
      class="post-content"
    >
      <h2>
        title 1
         
        1
      </h2>
      <hr />
      <span>
        body 1
      </span>
    </div>
  </div>
  <div
    class="sc-bdnxRM bkXcNm post"
  >
    <img
      alt="title 2"
      src="img/img2"
    />
    <div
      class="post-content"
    >
      <h2>
        title 2
         
        2
      </h2>
      <hr />
      <span>
        body 2
      </span>
    </div>
  </div>
  <div
    class="sc-bdnxRM bkXcNm post"
  >
    <img
      alt="title 3"
      src="img/img3"
    />
    <div
      class="post-content"
    >
      <h2>
        title 3
         
        3
      </h2>
      <hr />
      <span>
        body 3
      </span>
    </div>
  </div>
</div>
`;
